pkg/controller/atlasconnection/atlasconnection_controller.go

```diff
@@ -0,0 +1,410 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	ptr "k8s.io/utils/pointer"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	dbaasv1alpha1 "github.com/RHEcosystemAppEng/dbaas-operator/api/v1alpha1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/dbaas"
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlas"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/customresource"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/watch"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	DBUserNameKey = "username"
+	DBPasswordKey = "password"
+
+	digits   = "0123456789"
+	specials = "~=+%^*/()[]{}/!@#$?|"
+	all      = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz" + digits + specials
+)
+
+// MongoDBAtlasConnectionReconciler reconciles a MongoDBAtlasConnection object
+type MongoDBAtlasConnectionReconciler struct {
+	Client    client.Client
+	Clientset *kubernetes.Clientset
+	watch.ResourceWatcher
+	Log             *zap.SugaredLogger
+	Scheme          *runtime.Scheme
+	AtlasDomain     string
+	GlobalAPISecret client.ObjectKey
+	EventRecorder   record.EventRecorder
+}
+
+// Dev note: duplicate the permissions in both sections below to generate both Role and ClusterRoles
+
+// +kubebuilder:rbac:groups=atlas.mongodb.com,resources=MongoDBAtlasConnections,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=atlas.mongodb.com,resources=MongoDBAtlasConnections/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
+
+// +kubebuilder:rbac:groups=atlas.mongodb.com,namespace=default,resources=MongoDBAtlasConnections,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=atlas.mongodb.com,namespace=default,resources=MongoDBAtlasConnections/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="",namespace=default,resources=secrets,verbs=get;list;watch
+
+func (r *MongoDBAtlasConnectionReconciler) Reconcile(cx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = cx
+	log := r.Log.With("MongoDBAtlasConnection", req.NamespacedName)
+
+	conn := &dbaas.MongoDBAtlasConnection{}
+	if err := r.Client.Get(cx, req.NamespacedName, conn); err != nil {
+		if errors.IsNotFound(err) {
+			// CR deleted since request queued, child objects getting GC'd, no requeue
+			log.Info("DBaaSInventory resource not found, has been deleted")
```

Comment:
can be changed dot `log.Info("MongoDBAtlasConnection resource not found, has been deleted")` ?

Reply:
Done